modular_zubbers/code/modules/alt_anomaly_refinery/anomaly_refinery.dm

```diff
@@ -0,0 +1,120 @@
+
+/obj/machinery/research/anomaly_refinery/examine(mob/user)
+	. = ..()
+	. += span_notice("Usage of the machine will increase the bomb range requirement for the next experiment, however this diminishes with time.")
+	. += span_notice("Requires <b>[display_power(REFINERY_ANOMALY_POWER_REQUIREMENT)]</b> of power in the network to use.")
+	. += span_notice("Each core refinement takes <b>[DisplayTimeText(REFINERY_ANOMALY_REFINEMENT_TIME,1)]</b> to complete.")
+
+/obj/machinery/research/anomaly_refinery/get_required_radius(anomaly_type)
+	var/explosion_range = max(GLOB.MAX_EX_DEVESTATION_RANGE,GLOB.MAX_EX_HEAVY_RANGE,GLOB.MAX_EX_LIGHT_RANGE,0)
+	var/explosion_mod = 0.25 + (requirement_mod/100)*0.75
+	return round(explosion_range * explosion_mod,1)
+
+/obj/machinery/research/anomaly_refinery/start_test()
+
+	if (active)
+		say("ERROR: Already running a compression test.")
+		return
+
+	if(!istype(inserted_core) || !istype(inserted_bomb))
+		end_test("ERROR: Missing equpment. Items ejected.")
+		return
+
+	if(!inserted_bomb?.tank_one || !inserted_bomb?.tank_two || !(tank_to_target == inserted_bomb?.tank_one || tank_to_target == inserted_bomb?.tank_two))
+		end_test("ERROR: Transfer valve malfunctioning. Items ejected.")
+		return
+
+	if(!use_power_from_net(REFINERY_ANOMALY_POWER_REQUIREMENT))
+		say("ERROR: Not enough power in network to safely compress the core.")
+		return
+
+	say("Beginning compression test. Opening transfer valve.")
+	active = TRUE
+	test_status = null
+
+	inserted_bomb.toggle_valve(tank_to_target)
+
+	requirement_mod = min(requirement_mod + 20, 100) + round(REFINERY_ANOMALY_REFINEMENT_TIME/5,1)
+
+	timeout_timer = addtimer(CALLBACK(src, PROC_REF(timeout_test)), REFINERY_ANOMALY_REFINEMENT_TIME, TIMER_STOPPABLE | TIMER_UNIQUE | TIMER_DELETE_ME | TIMER_NO_HASH_WAIT)
+	requirement_timer = addtimer(CALLBACK(src, PROC_REF(remove_requirement)), 5 SECONDS, TIMER_UNIQUE | TIMER_STOPPABLE | TIMER_DELETE_ME | TIMER_NO_HASH_WAIT | TIMER_LOOP)
+
+	say("Refining core. ETA [DisplayTimeText(REFINERY_ANOMALY_REFINEMENT_TIME,1)] until core refinement completion.")
+
+	playsound(
+		src,
+		pick('sound/effects/explosion1.ogg','sound/effects/explosion2.ogg','sound/effects/explosion3.ogg'),
+		50
+	)
+
+	return
+
+/obj/machinery/research/anomaly_refinery/proc/remove_requirement()
+
+	requirement_mod = max(0,requirement_mod-1)
+
+	if(requirement_mod <= 0)
+		deltimer(requirement_timer)
+		requirement_timer = null
+		playsound(
+			src,
+			pick('modular_zubbers/sound/arcade/minesweeper_boardpress.ogg'),
+			50
+		)
+		say("Anomaly stability now at 100%.")
+
+/obj/machinery/research/anomaly_refinery/check_test(atom/source, list/arguments)
+
+	if(!inserted_core)
+		test_status = "ERROR: No core present during detonation."
+		playsound(
+			src,
+			pick('modular_zubbers/sound/arcade/minesweeper_winfail.ogg'),
+			50
+		)
+		return COMSIG_CANCEL_EXPLOSION
+
```

Comment:
Might be easier to just do . = COMSIG_CANCEL_EXPLOSION at the start of the proc and use empty returns

```suggestion
		)
		return

```